R/fitNullModel_internal.R

```diff
@@ -44,58 +44,61 @@
         }
         if (!is.null(covMatList)){
             if (is.null(group.idx)) group.idx <- list(resid.var = 1:length(y))
-            vc.mod <- .runAIREMLgaussian(y, X, start = start, covMatList = covMatList, 
-                                         group.idx = group.idx, AIREML.tol = AIREML.tol, drop.zeros = drop.zeros,  
+            vc.mod <- .runAIREMLgaussian(y, X, start = start, covMatList = covMatList,
+                                         group.idx = group.idx, AIREML.tol = AIREML.tol, drop.zeros = drop.zeros,
                                          max.iter = max.iter, EM.iter = EM.iter, verbose = verbose)
-            out <- .nullModOutMM(y = y, workingY = y, X = X, vc.mod = vc.mod, 
-                                 family = family, covMatList = covMatList, 
+            out <- .nullModOutMM(y = y, workingY = y, X = X, vc.mod = vc.mod,
+                                 family = family, covMatList = covMatList,
                                  group.idx = group.idx, drop.zeros = drop.zeros)
         }
-    } 
+    }
 
     ### Non-Gaussian family
     if (family$family != "gaussian"){
         # initial fit with glm
         mod <- glm(y ~ X, family = family)
-        
-        if (!is.null(covMatList)){ ## iterate between computing workingY and estimating VCs. 
-            iterate.out <- .iterateAIREMLworkingY(glm.mod = mod, X = X, family = family, 
+
+        if (!is.null(covMatList)){ ## iterate between computing workingY and estimating VCs.
+            iterate.out <- .iterateAIREMLworkingY(glm.mod = mod, X = X, family = family,
                                                   start = start, covMatList = covMatList, AIREML.tol = AIREML.tol,
                                                   drop.zeros = drop.zeros, max.iter = max.iter, EM.iter = EM.iter,
                                                   verbose = verbose)
-            
+
             vc.mod <- iterate.out$vc.mod
             working.y <- iterate.out$working.y
-            
+
       	    ## check whether all variance components were estimated as zero:
             if (vc.mod$allZero == TRUE){
                 out <- .nullModOutReg(y, X, mod, family)
                 out$zeroFLAG <- TRUE
             } else{
-                out <- .nullModOutMM(y = y, workingY = working.y$Y, X = X, vc.mod = vc.mod, 
-                                     family = family, covMatList = covMatList, 
+                out <- .nullModOutMM(y = y, workingY = working.y$Y, X = X, vc.mod = vc.mod,
+                                     family = family, covMatList = covMatList,
                                      vmu = working.y$vmu, gmuinv = working.y$gmuinv, drop.zeros = drop.zeros)
-            }	
+            }
         } else{
             out <- .nullModOutReg(y, X, mod, family)
         }
     }
 
     out.class <- class(out)
     nullprep <- nullModelTestPrep(out)
-    out <- c(out, nullprep)
+    # Add to fit data frame.
```

Comment:
In line 16 above (can't leave a comment there), I think you could use your new covMat naming function

Reply:
Thanks, I missed that one.